src/robotFactory.js

```diff
@@ -1,10 +1,72 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    const coordsZero = {
+      x: 0,
+      y: 0,
+    };
```

Comment:
```suggestion
    const { x = 0, y = 0 } = coords;
```
Instead of creating a new object, you can use destructuring with coords and give them a default value in case there are no `coords`, and then just:

```
this.coords = {
      x,
      y,
    };
```